ee/rbac/handler_test.go

```
@@ -0,0 +1,160 @@
+func newHandlerRig(handler *RBACHandler) func(t *testing.T, method, path, body string, principal *services.DashboardPrincipal) *httptest.ResponseRecorder {
+	router := mux.NewRouter()
+	router.HandleFunc("/roles", handler.ListRolesHandler).Methods(http.MethodGet)
+	router.HandleFunc("/roles", handler.CreateRoleHandler).Methods(http.MethodPost)
+	router.HandleFunc("/roles/{ROLE_ID}", handler.UpdateRoleHandler).Methods(http.MethodPut)
+	router.HandleFunc("/roles/{ROLE_ID}", handler.DeleteRoleHandler).Methods(http.MethodDelete)
+	router.HandleFunc("/users/{USER_ID}/grants", handler.GetUserGrantsHandler).Methods(http.MethodGet)
+	router.HandleFunc("/users/{USER_ID}/grants", handler.SetUserGrantsHandler).Methods(http.MethodPut)
+	router.HandleFunc("/me/permissions", handler.GetMyPermissionsHandler).Methods(http.MethodGet)
+	return func(t *testing.T, method, path, body string, principal *services.DashboardPrincipal) *httptest.ResponseRecorder {
+		t.Helper()
+		req := httptest.NewRequest(method, path, strings.NewReader(body))
```

golangci-lint (2.12.2)

[error] net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext

(noctx)